packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterWebView.java

```diff
@@ -341,65 +349,49 @@ public void onReceiveValue(String value) {
   }
 
   @SuppressWarnings("unchecked")
-  private void addJavaScriptChannels(MethodCall methodCall, Result result) {
-    List<String> channelNames = (List<String>) methodCall.arguments;
-    registerJavaScriptChannelNames(channelNames);
-    result.success(null);
+  private void addJavaScriptChannels(WebView webView, List<String> channelNames) {
+    registerJavaScriptChannelNames(webView, channelNames);
   }
 
   @SuppressWarnings("unchecked")
-  private void removeJavaScriptChannels(MethodCall methodCall, Result result) {
-    List<String> channelNames = (List<String>) methodCall.arguments;
+  private void removeJavaScriptChannels(WebView webView, List<String> channelNames) {
     for (String channelName : channelNames) {
       webView.removeJavascriptInterface(channelName);
     }
-    result.success(null);
   }
 
-  private void clearCache(Result result) {
+  private void clearCache(WebView webView) {
     webView.clearCache(true);
     WebStorage.getInstance().deleteAllData();
-    result.success(null);
   }
 
-  private void getTitle(Result result) {
-    result.success(webView.getTitle());
+  private String getTitle(WebView webView) {
+    return webView.getTitle();
   }
 
-  private void scrollTo(MethodCall methodCall, Result result) {
-    Map<String, Object> request = methodCall.arguments();
-    int x = (int) request.get("x");
-    int y = (int) request.get("y");
-
+  private void scrollTo(WebView webView, int x, int y) {
     webView.scrollTo(x, y);
-
-    result.success(null);
   }
 
-  private void scrollBy(MethodCall methodCall, Result result) {
-    Map<String, Object> request = methodCall.arguments();
-    int x = (int) request.get("x");
-    int y = (int) request.get("y");
-
+  private void scrollBy(WebView webView, int x, int y) {
     webView.scrollBy(x, y);
-    result.success(null);
   }
 
-  private void getScrollX(Result result) {
-    result.success(webView.getScrollX());
+  private int getScrollX(WebView webView) {
+    return webView.getScrollX();
   }
 
-  private void getScrollY(Result result) {
-    result.success(webView.getScrollY());
+  private int getScrollY(WebView webView) {
+    return webView.getScrollY();
   }
 
-  private void applySettings(Map<String, Object> settings) {
+  private void applySettings(WebView webView, Map<String, Object> settings) {
```

Comment:
When marking the `updateJsMode` method as `static`, this method can also be marked `static`. This would have a small performance advantage and also guards against usage of instance members.